packages/babel-generator/src/printer.js

```diff
@@ -367,7 +368,7 @@ export default class Printer {
 
     const loc = t.isProgram(node) || t.isFile(node) ? null : node.loc;
     this.withSource("start", loc, () => {
-      this[node.type](node, parent);
+      printMethod(node, parent);
```

Comment:
We could do
```suggestion
      printMethod.call(this, node, parent);
```

to get it working.